src/scheduler/MpiWorld.cpp

```diff
@@ -1,20 +1,21 @@
-#include <faabric/mpi/mpi.h>
-
-#include <faabric/scheduler/MpiThreadPool.h>
 #include <faabric/scheduler/MpiWorld.h>
 #include <faabric/scheduler/Scheduler.h>
-#include <faabric/transport/MpiMessageEndpoint.h>
 #include <faabric/util/environment.h>
 #include <faabric/util/func.h>
 #include <faabric/util/gids.h>
-#include <faabric/util/logging.h>
 #include <faabric/util/macros.h>
-#include <faabric/util/timing.h>
 
-static thread_local std::unordered_map<int, std::future<void>> futureMap;
-static thread_local std::unordered_map<std::string,
-                                       faabric::scheduler::FunctionCallClient>
-  functionCallClients;
+/* Each MPI rank runs in a separate thread, however they interact with faabric
+ * as a library. Thus, we use thread_local storage to guarantee that each rank
+ * sees its own version of these data structures.
```

Comment:
I don't quite understand this comment, what do you mean "they interact with faabric as a library"?